### ISSUE TITLE
[FEAT]: Import files from extracted url

### ISSUE THREAD

**Issue by pascaleproulx** [DARPA-ASKEM/Terarium#1142]
**Is your feature request related to a problem? Please describe.**
Currently, if we use the feature to upload a file from a github url, it loads the file as code, even if it is a csv (i.e. a dataset).

**Describe the solution you'd like**
The current github repo browser should be updated to this:
![image](https://github.com/DARPA-ASKEM/Terarium/assets/42645294/5ba2da02-6fb2-4f96-8a34-5a6f68b5f7cf)
where:
- [X] The files are in a multi-select list
- [X] The files are group by type based on their extensions
- [X] Code (.js, .py, .m, etc)
- [X] Data (.csv, .json, eyc)
- [X] Documents (.md, .txt, .pdf, etc)
- [X] Unknown file type (anything not falling in the above categories). 
- [ ] User could select from the above three types using the dropdown and bring that content in. - Note: This part could be done in a later pass. **DEV NOTE: this is going to be done in another pass**
- [X] The selected/file in focus is show in the preview window. Note: this could be done as a second pass.  **DEV NOTE: Text files only are previewed - PDFs are not, this will be done in another pass**
- [X] The Import bottom should indicate how many files are selected/will be imported.
- [ ] The end result should be to see all these files in the Resources panel in their proper bin.  **DEV NOTE: Blocked by inability to save files from github in current version TDS. This will be done in another pass**
- [ ] Provenance should show under the preview panel

**Comment by jamiewaese-uncharted** [comment]
A small tweak: Prime has a "tree" component we can use to show folders and sub folders

![importing from GitHub 2 with folders](https://github.com/DARPA-ASKEM/Terarium/assets/120480244/439c21bb-7647-4357-84ba-5c6fe160ef71)

**Comment by jamiewaese-uncharted** [comment]
If there are folders and sub folder, we can have a breadcrumb along the top, and a section for "Folders", which if a user clicks on one, we drill down into it. 
![image](https://github.com/DARPA-ASKEM/Terarium/assets/120480244/0152ae35-ba68-46d1-bbe1-238eac196b58)

**Comment by jamiewaese-uncharted** [comment]
One more thing.... we can add a section beneath the preview to show the provenance information that will be associated with each of the files that are imported.
![image](https://github.com/DARPA-ASKEM/Terarium/assets/120480244/43303ca1-eb61-46ad-94a5-7dc641f0e353)

**Comment by dvince2** [comment]
@jamiewaese-uncharted re: provenance, I will open a new issue to tackle that one.  In this case, would the provenance all the same as whats there? Or are we pulling this from somewhere?

**Comment by dvince2** [comment]
Breakoff issues:
unknown types: https://github.com/DARPA-ASKEM/Terarium/issues/1164
dataset creation: https://github.com/DARPA-ASKEM/Terarium/issues/1165
document creation: https://github.com/DARPA-ASKEM/Terarium/issues/1166
code creation: https://github.com/DARPA-ASKEM/Terarium/issues/1167
provenance: https://github.com/DARPA-ASKEM/Terarium/issues/1168
